src/SprykerEco/Zed/AfterPay/Communication/Plugin/Checkout/AfterPaySaveOrderPlugin.php

```diff
@@ -34,6 +38,16 @@ class AfterPaySaveOrderPlugin extends AbstractPlugin implements CheckoutDoSaveOr
      */
     public function saveOrder(QuoteTransfer $quoteTransfer, SaveOrderTransfer $saveOrderTransfer): void
     {
+        if ($quoteTransfer->getPayment()->getPaymentProvider() !== SharedAfterPayConfig::PROVIDER_NAME) {
+            return;
+        }
+
         $this->getFacade()->saveOrderPayment($quoteTransfer, $saveOrderTransfer);
+
+        $afterPayCallTransfer = $this->getFactory()
+            ->createQuoteToCallConverter()
+            ->convert($quoteTransfer);
+
+        $this->getFacade()->authorizePayment($afterPayCallTransfer);
```

Comment:
You did 2 facade calls that violate spryker conventions, so I suggest moving all business logic into one facade method.